components/camel-cdi-jee/src/main/java/org/apache/camel/cdi/transaction/JavaEETransactionPolicy.java

```diff
@@ -0,0 +1,136 @@
+package org.apache.camel.cdi.transaction;
+
+import javax.annotation.Resource;
+import javax.transaction.TransactionManager;
+
+import org.apache.camel.Processor;
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.builder.ErrorHandlerBuilder;
+import org.apache.camel.builder.ErrorHandlerBuilderRef;
+import org.apache.camel.model.ProcessorDefinition;
+import org.apache.camel.spi.RouteContext;
+import org.apache.camel.spi.TransactedPolicy;
+import org.apache.camel.util.ObjectHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Sets a proper error handler. This class is based on
+ * {@link org.apache.camel.spring.spi.SpringTransactionPolicy}.
+ * <p>
+ * This class requires the resource {@link TransactionManager} to be available
+ * through JNDI url &quot;java:/TransactionManager&quot;
+ */
+public abstract class JavaEETransactionPolicy implements TransactedPolicy {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JavaEETransactionPolicy.class);
+
+    public static interface Runnable {
+        void run() throws Throwable;
+    }
+
+    @Resource(lookup = "java:/TransactionManager")
```

Comment:
We may want to refine the strategy to lookup for the transaction manager. In a _micro service_ (or MicroProfile), we may want to use CDI and JTA only. So the lookup will be done through CDI and not JNDI. That being said, we can refine later on.